faster_whisper/audio.py

```diff
@@ -29,91 +17,42 @@ def decode_audio(
       split_stereo: Return separate left and right channels.
 
     Returns:
-      A float32 Numpy array.
+      A float32 Torch Tensor.
 
       If `split_stereo` is enabled, the function returns a 2-tuple with the
       separated left and right channels.
     """
-    resampler = av.audio.resampler.AudioResampler(
-        format="s16",
-        layout="mono" if not split_stereo else "stereo",
-        rate=sampling_rate,
-    )
-
-    raw_buffer = io.BytesIO()
-    dtype = None
 
-    with av.open(input_file, mode="r", metadata_errors="ignore") as container:
-        frames = container.decode(audio=0)
-        frames = _ignore_invalid_frames(frames)
-        frames = _group_frames(frames, 500000)
-        frames = _resample_frames(frames, resampler)
-
-        for frame in frames:
-            array = frame.to_ndarray()
-            dtype = array.dtype
-            raw_buffer.write(array)
-
-    # It appears that some objects related to the resampler are not freed
-    # unless the garbage collector is manually run.
-    del resampler
-    gc.collect()
-
-    audio = np.frombuffer(raw_buffer.getbuffer(), dtype=dtype)
-
-    # Convert s16 back to f32.
-    audio = audio.astype(np.float32) / 32768.0
+    waveform, audio_sf = torchaudio.load(input_file)  # waveform: channels X T
```

Comment:
Should add supported audio formats in readme, or add function to convert unsupported audio formats

Reply:
I don't have information about unsupported formats, can you point me to any source that has information about them?

Reply:
Me too. But you can refer to this [link](https://modelzoo.co/model/audio#:~:text=mp3%2C%20wav%2C%20aac%2C%20ogg,other%20format%20supported%20by%20libsox.) for torchaudio supported formats. At least currently the m4a files in my examples are not running :(

Reply:
I just loaded `m4a` file without any issues. Can you examine the codec?

`ffprobe -v error -show_entries stream=codec_name -of default=noprint_wrappers=1:nokey=1 filename.m4a`

Reply:
`torchaudio` uses three libraries based on their availability in the environment, `soundfile` -> `sox` -> `ffmpeg`, so you should know which backend it's using that doesn't support `m4a`

Reply:
I received this error with m4a file:
```python
Traceback (most recent call last):
  File "/root/faster-whisper/infer.py", line 74, in <module>
    segments, info = batched_model.transcribe(audio_path, batch_size=16)
  File "/root/faster-whisper/faster_whisper/transcribe.py", line 526, in transcribe
    audio = decode_audio(audio, sampling_rate=sampling_rate)
  File "/root/faster-whisper/faster_whisper/audio.py", line 26, in decode_audio
    waveform, audio_sf = torchaudio.load(input_file)  # waveform: channels X T
  File "/root/miniconda3/envs/transformer/lib/python3.8/site-packages/torchaudio/_backend/utils.py", line 205, in load
    return backend.load(uri, frame_offset, num_frames, normalize, channels_first, format, buffer_size)
  File "/root/miniconda3/envs/transformer/lib/python3.8/site-packages/torchaudio/_backend/soundfile.py", line 27, in load
    return soundfile_backend.load(uri, frame_offset, num_frames, normalize, channels_first, format)
  File "/root/miniconda3/envs/transformer/lib/python3.8/site-packages/torchaudio/_backend/soundfile_backend.py", line 221, in load
    with soundfile.SoundFile(filepath, "r") as file_:
  File "/root/miniconda3/envs/transformer/lib/python3.8/site-packages/soundfile.py", line 658, in __init__
    self._file = self._open(file, mode_int, closefd)
  File "/root/miniconda3/envs/transformer/lib/python3.8/site-packages/soundfile.py", line 1216, in _open
    raise LibsndfileError(err, prefix="Error opening {0!r}: ".format(self.name))
soundfile.LibsndfileError: Error opening '/root/faster-whisper/tests/data/benchmark.m4a': Format not recognised.
```

Reply:
can you `pip install sox` and try again?

Reply:
I just tested again, `m4a` is indeed not supported using neither `soundfile` nor `sox` only `ffmpeg` backend supports it, so the solution is to instruct the users to install `ffmpeg` if their format isn't supported, or ( and I feel sorry for saying this :laughing: ) use PyAV for loading and keep the rest as is

Reply:
I think should change backend to ffmpeg to support more audio formats:
```python
waveform, audio_sf = torchaudio.load(input_file, backend="ffmpeg")
```
And add ffmpeg to the installation doc. Ex for linux:
```
sudo apt-get install ffmpeg
```

Reply:
this is implicitly done by `torchaudio`, if `ffmpeg` exists, it will be included and chosen automatically, if not, it will fall back to the next one which is `sox`, if not, it will fall back to the final one `soundfile` which must exist because it's included in `torchaudio` requirements